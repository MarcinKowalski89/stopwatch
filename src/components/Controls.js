import React from 'react';

class Controls extends React.Component {

  render() {
    return (
      <div className="d-flex justify-content-center btn-group">
        <button className="btn btn-success">Start</button>
        <button className="btn btn-danger">Stop</button>
        <button className="btn btn-warning">Reset</button>
        <button className="btn btn-primary">Save time to list</button>
      </div>
    );
  }

}

export default Controls;