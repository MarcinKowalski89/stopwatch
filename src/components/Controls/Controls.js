import React from 'react';
import PropTypes from 'prop-types';

export const Controls = (props) => {
  const {
    isRunning, start, stop, reset, save,
  } = props;
  return (
    <div className="d-flex justify-content-center btn-group">
      {
        isRunning
        ? <button className="btn btn-danger" onClick={stop}>Stop</button>
        : <button className="btn btn-success" onClick={start} disabled={isRunning}>Start</button>
      }
      <button className="btn btn-warning" onClick={reset}>Reset</button>
      <button className="btn btn-primary" onClick={save} disabled={!isRunning}>Save time to list</button>
    </div>
  );
};

Controls.propTypes = {
  isRunning: PropTypes.bool.isRequired,
  start: PropTypes.func.isRequired,
  stop: PropTypes.func.isRequired,
  reset: PropTypes.func.isRequired,
  save: PropTypes.func.isRequired,
};

export default Controls;
