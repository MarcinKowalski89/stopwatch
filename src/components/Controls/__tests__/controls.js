import React from 'react';
import { shallow } from 'enzyme';

import { Controls } from '../Controls';

test('Controls shoud match default snapshot', () => {
  const component = shallow(<Controls
    isRunning={false}
    start={() => null}
    stop={() => null}
    reset={() => null}
    save={() => null}
  />);

  expect(component).toMatchSnapshot();
});

test('Controls shoud match snapshot for running time', () => {
  const component = shallow(<Controls
    isRunning
    start={() => null}
    stop={() => null}
    reset={() => null}
    save={() => null}
  />);

  expect(component).toMatchSnapshot();
});
