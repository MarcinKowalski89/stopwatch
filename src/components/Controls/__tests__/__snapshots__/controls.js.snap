// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls shoud match default snapshot 1`] = `
<div
  className="d-flex justify-content-center btn-group"
>
  <button
    className="btn btn-success"
    disabled={false}
    onClick={[Function]}
  >
    Start
  </button>
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="btn btn-primary"
    disabled={true}
    onClick={[Function]}
  >
    Save time to list
  </button>
</div>
`;

exports[`Controls shoud match snapshot for running time 1`] = `
<div
  className="d-flex justify-content-center btn-group"
>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Stop
  </button>
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
  >
    Save time to list
  </button>
</div>
`;
